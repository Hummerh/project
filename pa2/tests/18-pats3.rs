fn main() {
      let mut y : i32 = 1;

      match (y) {
            ref x => { x; }
      };

      match (y) {
            ref mut x => { x; }
      };

      match (y) {
            mut x => { x; }
      };
}
