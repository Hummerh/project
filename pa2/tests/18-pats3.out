(crate
    (items
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-mut-id
                        (id
                            (y)))
                    (type-i32)
                    (lit-dec))
                (match
                    (id
                        (y))
                    (match-arms
                        (match-arm
                            (pats
                                (pat-ref-id
                                    (id
                                        (x))))
                            (block
                                (id
                                    (x))))))
                (match
                    (id
                        (y))
                    (match-arms
                        (match-arm
                            (pats
                                (pat-ref-mut-id
                                    (id
                                        (x))))
                            (block
                                (id
                                    (x))))))
                (match
                    (id
                        (y))
                    (match-arms
                        (match-arm
                            (pats
                                (pat-mut-id
                                    (id
                                        (x))))
                            (block
                                (id
                                    (x))))))))))